src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsnonassociationsapi/service/SyncAndMigrateService.kt

```diff
@@ -0,0 +1,95 @@
+package uk.gov.justice.digital.hmpps.hmppsnonassociationsapi.service
+
+import com.microsoft.applicationinsights.TelemetryClient
+import jakarta.persistence.EntityNotFoundException
+import org.slf4j.Logger
+import org.slf4j.LoggerFactory
+import org.springframework.stereotype.Service
+import org.springframework.transaction.annotation.Transactional
+import uk.gov.justice.digital.hmpps.hmppsnonassociationsapi.dto.CreateSyncRequest
+import uk.gov.justice.digital.hmpps.hmppsnonassociationsapi.dto.MigrateRequest
+import uk.gov.justice.digital.hmpps.hmppsnonassociationsapi.dto.NonAssociation
+import uk.gov.justice.digital.hmpps.hmppsnonassociationsapi.dto.UpdateSyncRequest
+import uk.gov.justice.digital.hmpps.hmppsnonassociationsapi.jpa.repository.NonAssociationsRepository
+
+@Service
+@Transactional
+class SyncAndMigrateService(
```

Comment:
can i double check, is the purpose of having a bespoke sync/migrate service, rather than reusing the normal service, basically to not publish events that would create an infinite sync loop?

Reply:
Yes and also we can delete this code afterwards